showQuestions.js

```diff
@@ -19,22 +19,27 @@ var converter = new showdown.Converter();
 async function showInstructions() {
   var elements = document.getElementsByTagName("div");
   var noteText = "Upload the questions before you hit start";
-  chrome.storage.local.get("questions", ({questions}) => {
-    if (questions && questions.length>2) {
-      noteText = "There is already a loaded set of questions in the browser cache. Re-upload if you want to be sure, and click start."
-    } 
-    for (item of elements) {
-      try {
-        if (item.className === "cell w99 h10") {
-          item.id = "42";
-          const instructionsElement = getInstructionsElement(noteText);
-          item.append(instructionsElement);
-        }
-      } catch (e) {}
+  const {questions} = await chrome.storage.local.get("questions");
+  if (questions && questions.length > 2) {
+    const {fileName} = await chrome.storage.local.get("fileName");
+    noteText = "There is already a loaded set of questions in the browser cache. ";
+    if (fileName !== "") {
+      noteText += `Question set name: "${fileName}". `;
     }
-  });
+    noteText += "Re-upload if you want to be sure, and click start.";
+  }
+  for (item of elements) {
+    try {
+      if (item.className === "cell w99 h10") {
+        item.id = "42";
+        const instructionsElement = getInstructionsElement(noteText);
+        item.append(instructionsElement);
+      }
+    } catch (e) {}
```

Comment:
Curious why you chose not to print out the catch block error message?

Reply:
I didn't want to change what the original author had coded, and also it was not the purpose of this PR anyway to do error handling.

Since `item`s are being inserted in to something existing in the DOM, there isn't much else you can do except `console.error` the exception, warn the user and throw it again. Maybe show whatever you wanted to show as an `alert` instead (eww).